[Swaminathan Viswa] iP

SwaminathanViswa opened the pull request:
### **BitBot**

Welcome to your most favourite and important chatbot which **HELPS** you to be super organised! It is super easy to learn and it is **FAST!!**

Here are some things it can do:

- [ ] - It can add / delete tasks
- [ ] - It can add events, deadlines and simple todo tasks
- [ ] - It can mark them as done or not done!!!!

All you need to do is:

1. download the .jar file from [here](https://github.com/SwaminathanViswa/ip/releases/tag/A-Jar)
2. Open the .jar file in your favourite IDE
3. And voila you get your own working chatbot!!!!! 😄 

Here is the `main` method and the code to run it on gradle:

```ruby
public class BitBot {
    public static void main(String[] args) throws BitBotException, FileNotFoundException {
        Application.run();
    }
}
```

If you are using Intellij and you have Gradle setup, you can use:

`gradle run` after double clicking "control" twice.
or
`./gradlew run` in the command line terminal

**There are more things coming up:**
- A GUI for easier communication
- Managing deadlines
- and many more to look out for!! 👍

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SwaminathanViswa committed:
Changed from DUKE to BitBot

SwaminathanViswa committed:
changed DUKE to BitBot

SwaminathanViswa committed:
added the scanner to read user input

SwaminathanViswa committed:
added the ArrayList to store events

SwaminathanViswa committed:
added the mark as done component

SwaminathanViswa committed:
added the toDos, events and deadlines

SwaminathanViswa committed:
added test cases and inputs for testing

SwaminathanViswa committed:
added exception handling

SwaminathanViswa committed:
added delete functionality

SwaminathanViswa committed:
added new tests to input.txt and corresponding expected output

SwaminathanViswa committed:
added javadocs and moved the common code into another method

SwaminathanViswa committed:
added comments

SwaminathanViswa committed:
Add saving feature

SwaminathanViswa committed:
Merge branch 'branch-Level-7'

SwaminathanViswa committed:
Add functionality to Dates and Times

SwaminathanViswa committed:
Merge branch 'branch-Level-8'

SwaminathanViswa committed:
Refactor the code to use more OOP

SwaminathanViswa committed:
Put classes into bitbot package

SwaminathanViswa committed:
Merge branch 'add-gradle-support'

SwaminathanViswa committed:
Set up Gradle for BitBot

SwaminathanViswa committed:
Add JUnit Tests for DeadlineTest.java, EventsTest.java, TaskTest.java,
ToDosTest.java

SwaminathanViswa committed:
Edit shadowJar names

SwaminathanViswa committed:
Add find functionality

SwaminathanViswa committed:
Merge branch 'branch-Level-9'

SwaminathanViswa committed:
Add javadocs to public classes/methods

SwaminathanViswa committed:
Merge branch 'branch-A-JavaDoc'

SwaminathanViswa committed:
Change alignment to suit coding standards

SwaminathanViswa committed:
Edit lines to ensure code standard

SwaminathanViswa committed:
Merge branch 'master' into branch-A-CodingStandard

# Conflicts:
#	src/main/java/bitbot/BitBot.java
#	src/main/java/bitbot/TaskHandling.java
#	src/main/java/bitbot/UI.java

SwaminathanViswa committed:
Merge branch 'branch-A-CodingStandard'

SwaminathanViswa committed:
Apend javadocs and edit lines to meet coding standard

SwaminathanViswa committed:
Reorganise to improve OOP principles

SwaminathanViswa committed:
Add checkstyle files and edit code accordingly

SwaminathanViswa committed:
Add varargs to handleFind method

SwaminathanViswa committed:
Merge branch 'A-Varargs'

sumomomomomo commented on src/main/java/bitbot/Application.java:
```diff
@@ -0,0 +1,162 @@
+import java.util.Scanner;
+
+public class Application {
+    private static final String PATH_TO_FILE = "./data/Bitbot.txt";
```
I like your use of screaming snake case for final constants like relative filepath.

sumomomomomo commented on src/main/java/bitbot/Application.java:
```diff
@@ -0,0 +1,162 @@
+                if (partsOfInput.length > 0) {
+                    keyWord = partsOfInput[0];
+
+                    switch (keyWord) {
```
I also just found out, but according to our coding standard, there should be no indentation for case clauses.

sumomomomomo commented on src/main/java/bitbot/Deadline.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    /**
+     * prints out the final display of the task details to the user.
```
Consider capitalizing the first letter of the opening sentence of each header comment for consistency. Same for description of return.

sumomomomomo commented on src/main/java/bitbot/TaskHandling.java:
```diff
@@ -0,0 +1,379 @@
+
+import java.util.ArrayList;
+
+public class TaskHandling {
```
Shouldn't "TaskHandling" be "TaskHandler"? Class names should be nouns.

sumomomomomo commented on src/main/java/bitbot/ToDos.java:
```diff
@@ -0,0 +1,31 @@
+
+import java.time.format.DateTimeFormatter;
+
+public class ToDos extends Task {
```
Consider renaming "ToDos" to "Todo": abbreviations aren't capitalized so it makes more sense for lowercase "d" in my opinion, and I feel singular "Todo" is more accurate to the class's function than plural "Todos".

limjeremy496 replied:
I concur with sumomomomomo! I feel that "Todo" would be more suited.

sumomomomomo reviewed (commented):
LGTM, but some cosmetic/naming changes would make it better!

SwaminathanViswa committed:
Add GUI to chatbot

SwaminathanViswa committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	src/main/java/bitbot/Application.java
#	src/main/java/bitbot/TaskHandling.java

edit merge conflicts

SwaminathanViswa committed:
Merge tag 'A-CheckStyle'

# Conflicts:
#	data/Bitbot.txt
#	src/main/java/bitbot/Application.java
#	src/main/java/bitbot/BitBot.java
#	src/main/java/bitbot/TaskHandling.java

SwaminathanViswa committed:
Edit merging mistakes

limjeremy496 commented on src/test/java/bitbot/ToDosTest.java:
```diff
@@ -0,0 +1,32 @@
+
+    Task toDos = new ToDos("Do Homework");
+
+    @Test
```
I like that the test method names are clear and descriptive, which helps in understanding the test's purpose.

limjeremy496 commented on src/main/java/bitbot/Storage.java:
```diff
@@ -0,0 +1,165 @@
+                    } catch (DateTimeParseException e) {
+                        try {
+                            LocalDate localDate = LocalDate.parse(deadlineBy, dateFormatter);
+                            task = new Deadline(descriptionOfDeadline, localDate);
```
can the nested try-catch be improved?

limjeremy496 commented on src/main/java/bitbot/UI.java:
```diff
@@ -0,0 +1,46 @@
+/**
+ * This UI class is to store the intro and the conclusions for the user interface.
+ */
+public class UI {
```
Should the class name be "Ui" instead?

limjeremy496 commented on src/main/java/bitbot/Events.java:
```diff
@@ -0,0 +1,122 @@
+ * This is the Events class which extends Task
+ * as this is where new Events objects are created
+ */
+public class Events extends Task {
```
Should the class name be "Event" instead?

limjeremy496 reviewed (commented):
Overall very great job! Code is very readable and logical :)

SwaminathanViswa committed:
Edit after comments from peer review

SwaminathanViswa committed:
Edit after comments from peer review

SwaminathanViswa committed:
Add assert statements to ensure correctness

SwaminathanViswa committed:
Merge pull request #1 from SwaminathanViswa/branch-A-Assertions

Add assert statements to ensure correctness

SwaminathanViswa committed:
Merge branch 'master' of https://github.com/SwaminathanViswa/ip

SwaminathanViswa committed:
Merge branch 'master' into branch-A-CodeQuality

SwaminathanViswa committed:
The codebase has methods related to input parsing that is
currently all in one class, making it difficult to read and understand.

This structure increases confusion and uses a lot of code duplication.

Rearrange this methods into a new Parser class to improve code
readability and code quality.

Creating a new Parser class ensures that all the input parsing methods
are in one place making it easier to read and understand.

SwaminathanViswa committed:
Merge pull request #2 from SwaminathanViswa/branch-A-CodeQuality

Branch-A-CodeQuality

SwaminathanViswa committed:
Merge branch 'master' of https://github.com/SwaminathanViswa/ip

SwaminathanViswa committed:
Merge branch 'master' into branch-A-Streams

SwaminathanViswa committed:
The methods extractTimeDetail and extractDescription has for loops and
conditional statements to get the respective details.

The current implementation is harder to read and it is not as flexible
as compared to Java streams.

Change the two methods to use Java Streams for processing and getting
the necessary details.

Streams provide a variety of options and it makes the code more
readable and easier to maintain.

SwaminathanViswa committed:
Merge pull request #3 from SwaminathanViswa/branch-A-Streams

Branch-A-Streams

SwaminathanViswa committed:
Merge branch 'master' of https://github.com/SwaminathanViswa/ip

SwaminathanViswa committed:
Add yml file to set up CI

SwaminathanViswa committed:
Add tagging function to Task.

The application does not have a functionality to tag tasks
to make it easier for identification and categorize them.

Users need a method to tag their tasks that they want.

Implement 'handleTag' and 'handleUntag' methods that allows users to
tag and untag tasks respectively.

This approach allows for proper input parsing and error handling.

SwaminathanViswa committed:
Merge pull request #4 from SwaminathanViswa/branch-extensionMain

Add tagging function to Task.

SwaminathanViswa committed:
Merge branch 'master' of https://github.com/SwaminathanViswa/ip

SwaminathanViswa committed:
Add JUnit tests for Parser class

SwaminathanViswa committed:
Merge branch 'master' into branch-A-personality

SwaminathanViswa committed:
Edit phrases used by Bitbot for unique personality

SwaminathanViswa committed:
Merge branch 'branch-A-personality'

SwaminathanViswa committed:
Add more JUnit Test cases for Storage class

SwaminathanViswa committed:
Refactor code to adopt SLAP principles

SwaminathanViswa committed:
Change code in handleTag to include more error handling cases
and refactor Storage.java to adhere to SLAP principles

SwaminathanViswa committed:
Edit README.md file and credit author

SwaminathanViswa committed:
Update README.md file

SwaminathanViswa committed:
Update README.md file

SwaminathanViswa committed:
Edit code to handle more errors and give credits

SwaminathanViswa committed:
Edit README.md to change image

SwaminathanViswa committed:
Add table to README.md and solve checkstyle issues

SwaminathanViswa committed:
Add break statement to case block

SwaminathanViswa committed:
Update README.md

SwaminathanViswa committed:
Update README.md
